app/services/bulk_upload/lettings/year2023/row_parser.rb

```diff
@@ -535,13 +543,15 @@ def validate_nulls
       if setup_question?(question)
         fields.each do |field|
           if errors[field].present?
-            errors.add(field, I18n.t("validations.not_answered", question: question.check_answer_label&.downcase), category: :setup)
+            question_text = question.check_answer_label.presence || question.header
+            errors.add(field, I18n.t("validations.not_answered", question: question_text&.downcase), category: :setup)
```

Comment:
Do we want any tests for this?

Reply:
@kosiakkatrina does the test at `spec/services/bulk_upload/lettings/year2023/row_parser_spec.rb:240` (see below) not test this functionality? Or do you mean adding another test but for the setup questions (the test mentioned deals with non-setup bus). Thanks!

Reply:
@bibblobcode not currently, no. My understanding is that I wouldn't be able to add that nil-handling logic to `en.yml` itself, instead I could update the code in this file so that if `question_text` is nil then I just pass "this question" or something to `validations.not_answered`. Does that sound sensible to you? Thanks!

Reply:
I meant a test for a setup section, it makes sense that this way of adding error is here, but I'm not a 100% sure if we have any cases where this would actually trigger. Happy for you to leave as is tho

Reply:
@kosiakkatrina I've been trying to write an equivalent test for the setup section but am struggling a bit. All setup questions have a `check_answer_label` so the ` || question.header` logic is not being invoked for any existing questions. Therefore I was trying to use something like `allow().to receive().and_return(nil)' to make it seem like one of the existing questions actually didn't have a `check_answer_label`. Does this sound like the right approach to take, and if so how would we fill in the arguments for `allow` in `allow().to receive().and_return(nil)`? I've had a think and am a bit stumped!

Before we continue it's worth saying this might not be worth investing any more time in - I'm really curious about what would be the right thing to do here, but I'm also happy to just leave it if it's not worth it!